python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -179,18 +183,22 @@ def _makeRecords(dataIds: Mapping[str, Iterable], universe: DimensionUniverse) -
 
     # Start populating dicts that will become DimensionRecords by providing
     # alternate keys like detector names
-    record_dicts_by_dimension_name = {}
+    record_dicts_by_dimension_name: dict[str, list[dict[str, str | int | bytes]]] = {}
     for name, values in complete_data_id_values.items():
         record_dicts_by_dimension_name[name] = []
-        dimension = universe[name]
+        dimension_el = universe[name]
```

Comment:
This change breaks obs_base so I think mypy is getting a bit confused as to what the return type of `universe[name]` is. I need to add a test in daf_butler like the one in obs_base so I can catch this early.